--- a/net/minecraft/client/render/entity/RenderPlayer.java
+++ b/net/minecraft/client/render/entity/RenderPlayer.java
@@ -38,8 +38,8 @@
 		if((itemStack6 = entityPlayer5.inventory.armorInventory[i4]) != null && (item7 = itemStack6.getItem()) instanceof ItemArmor) {
 			ItemArmor itemArmor8 = (ItemArmor)item7;
 			this.loadTexture("/armor/" + armorFilenamePrefix[itemArmor8.renderIndex] + "_" + (i3 == 2 ? 2 : 1) + ".png");
-			ModelBiped modelBiped9;
-			(modelBiped9 = i3 == 2 ? this.modelArmor : this.modelArmorChestplate).bipedHead.showModel = i3 == 0;
+			ModelBiped modelBiped9 = i3 == 2 ? this.modelArmor : this.modelArmorChestplate;
+			modelBiped9.bipedHead.showModel = i3 == 0;
 			modelBiped9.bipedHeadwear.showModel = i3 == 0;
 			modelBiped9.bipedBody.showModel = i3 == 1 || i3 == 2;
 			modelBiped9.bipedRightArm.showModel = i3 == 1;
